### DISCUSSION TITLE
Encoding format?

### DISCUSSION THREAD

**Discussion by bradh** [stac-extensions/video#10] [Ideas]
We have `video:codec_name`, but there are a range of encoding formats (e.g. is it MPEG-2 Transport Stream, MXF, something more like ISO Base Media File Format e.g. CMAF with ST 1910 conventions, NITF with MIE4NITF  extensions?). Not all formats are equally useful as search results.

Possibly have an enumerated list of formats, and allow registration of new ones?

**Comment by darrenwiens** [comment]
`ffmpeg probe` returns something like:

```
{'index': 0,
 'codec_name': 'h264',
 'codec_long_name': 'H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10',
 'profile': 'Baseline',
 'codec_type': 'video',
 'codec_tag_string': '[27][0][0][0]',
 'codec_tag': '0x001b',
...
```

I'm not completely attached to the idea of this property, but seems useful for a consumer to know up front.

My preference would be to point at a definitive list rather than maintaining an enumeration, if such a thing exists.

**Reply by bradh** [reply]
I assume that came out of a transport stream (i.e. MPEG-2). 0x1B is the stream_type for H.264 in that format.

FourCC didn't really come into vogue until later - its an MPEG-4 thing for box kinds.

I agree with pointing to another list, but not sure there is a good one to point to.

**Reply by bradh** [reply]
Adding the `profile` information might also be useful as an optional property.
